httpie/core.py

```diff
@@ -93,6 +91,13 @@ def main(
                 f'Too many redirects'
                 f' (--max-redirects={parsed_args.max_redirects}).'
             )
+        except requests.ConnectionError:
```

Comment:
@jakubroztocil Okay, we were able to move the code over as suggested. Should we create unique strings for each type of connection error or is the current general message that we implemented sufficient?

Reply:
Now it prints both the old and new error message:

```bash
$ http xxx.example.org
```

`http: error: Connection aborted while doing a GET request to URL: http://xxx.example.org/.`

`http: error: ConnectionError: HTTPConnectionPool(host='xxx.example.org', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x105ace990>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known'))`

This should be easy to fix.

Reply:
> Should we create unique strings for each type of connection error or is the current general message that we implemented sufficient?

When I look at old message, it is admittedly very verbose but it does say what the actual error was (`nodename nor servname provided, or not known` is the important bit).

The new one, `Connection aborted …`, doesn’t capture the actual reason and is even possibly misleading: as the prerequisite hostname resolving step failed, no connection was created, so it couldn’t have been aborted.

I’d definitely work on the text of the messages (it’s better to be overly verbose than misleading). 

If the `ConnectionError` always means the same thing, then it’s okay if your craft and hard-code a good error message just for this and move on. 

But it is possible that the exception error type alone is not sufficient, because the parent exception (`Caused by <parent> …`) might carry important context. 

(That’s why trying to trigger different errors and catalog them feels like an important step in developing understanding of the problem, which would allow you to come up with an elegant solution.)